modules/nf-core/ncbidatasets/download/main.nf

```diff
@@ -0,0 +1,108 @@
+process NCBIDATASETS_DOWNLOAD {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "conda-forge::ncbi-datasets-cli=15.11.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ncbi-datasets-pylib:15.11.0--pyhdfd78af_0':
+        'staphb/ncbi-datasets:15.11.0' }"
+
+    input:
+
+    val(meta)
+
+    output:
+    // Required output
+    tuple val(meta), path("*.zip")   ,             emit: zip
+    path "versions.yml"              ,             emit: versions
+    // Optional outputs, depends on tool usage
+    tuple val(meta), path("cds.fna.gz"),           emit: cds,        optional: true
+    tuple val(meta), path("gene.fna.gz"),          emit: gene,       optional: true
+    tuple val(meta), path("*genomic.fna.gz"),      emit: genome,     optional: true
+    tuple val(meta), path("genomic.gbff.gz"),      emit: gbff,       optional: true
+    tuple val(meta), path("genomic.gff.gz"),       emit: gff,        optional: true
+    tuple val(meta), path("genomic.gtf.gz"),       emit: gtf,        optional: true
+    tuple val(meta), path("protein.faa.gz"),       emit: protein,    optional: true
+    tuple val(meta), path("rna.fna.gz"),           emit: rna,        optional: true
+    tuple val(meta), path("3p_utr.fna.gz"),        emit: utr_3p,     optional: true
+    tuple val(meta), path("5p_utr.fna.gz"),        emit: utr_5p,     optional: true
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def valid_commands = ["genome accession", "genome taxon", "gene accession", "gene gene-id", "gene symbol", "gene taxon", "virus genome taxon", "virus genome accession", "virus protein"]
+    if (!valid_commands.contains(meta.command)) {
+        error "Unsupported command: ${meta.command} "
+    }
+
+
+    def args = task.ext.args ?: ''
+    def args_from_csv = meta.extra_args ?: ''
+    args_from_csv = args_from_csv.replaceAll("^\"|\"\$", "")
+    def prefix = task.ext.prefix ?: "${meta.id.replaceAll(' ', '_')}"
+
+    // In case of downloading based on taxon identifiers:
+    // multiple genomes/proteomes etc may exist in different files in the ncbi .zip file
+    // Therefore, only emit .zip file and do not emit extracted files
+    if (meta.command =~ /taxon/)
+        """
+
+        [ -e /usr/local/ssl/cacert.pem ] && export SSL_CERT_FILE=/usr/local/ssl/cacert.pem
```

Comment:
@nf-core/maintainers does anyone know whether this works across different infrastructures?

Reply:
What specifically are you referring to by 'this'?
The `[ -e <file> ]` is the same as `test -e <file>` which tests if a file exists.
If that's true (`&&`) then export the env. This is standard shell.

Reply:
Or is the question whether this file is user supplied or supplied by a container?

Reply:
This should be fine. It's kind of like how container specific files in busco are handled. If a file exists, do specific stuff related to it. The second part only happens if the first is true.